skbio/stats/composition.py

```diff
@@ -896,9 +948,27 @@ def ancom(table, grouping,
         else:
             nu = cutoff[4]
         reject = (W >= nu*n_feat)
-    labs = mat.columns
-    return pd.DataFrame({'W': pd.Series(W, index=labs),
-                         'reject': pd.Series(reject, index=labs)})
+
+    feat_ids = mat.columns
+    ancom_df = pd.DataFrame({'W': pd.Series(W, index=feat_ids),
+                             'reject': pd.Series(reject, index=feat_ids)})
+
+    if len(percentiles) == 0:
+        return ancom_df, pd.DataFrame()
+    else:
```

Comment:
Let me think about this one a little more this morning. I agree that this would be ideal (and it's the plan for the future as we discussed in #1293), but it's really important that we don't duplicate any logic for splitting samples into groups that could get out of sync. In the future, the splitting would be a operation performed by the feature/contingency table object that is provided to this function.